git/operations_test.go

```diff
@@ -59,24 +62,89 @@ func TestChangedFiles_NoPath(t *testing.T) {
 	}
 }
 
-func createRepo(dir string, nestedDir string) error {
-	fullPath := path.Join(dir, nestedDir)
-	var err error
-	if err = execCommand("git", "-C", dir, "init"); err != nil {
-		return err
+func TestOnelinelog_NoGitpath(t *testing.T) {
+	newDir, cleanup := testfiles.TempDir(t)
+	defer cleanup()
+
+	subdirs := []string{"dev", "prod"}
+
+	err := createRepo(newDir, subdirs)
+	if err != nil {
+		t.Fatal(err)
 	}
-	if err := execCommand("mkdir", "-p", fullPath); err != nil {
-		return err
+
+	if err = updateDirAndCommit(newDir, "dev", testfiles.FilesUpdated); err != nil {
+		t.Fatal(err)
 	}
-	if err = testfiles.WriteTestFiles(fullPath); err != nil {
-		return err
+	if err = updateDirAndCommit(newDir, "prod", testfiles.FilesUpdated); err != nil {
+		t.Fatal(err)
 	}
-	if err = execCommand("git", "-C", dir, "add", "--all"); err != nil {
-		return err
+
+	commits, err := onelinelog(newDir, "HEAD~2..HEAD", "")
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	fmt.Printf(">>> commits: %+v\n", commits)
```

Comment:
`go test -v` will include this output, which can add to the noise. Consider only printing diagnostic messages when things have gone wrong.

Reply:
A forgotten debug :(